--- qtstalker-0.26/lib/ChartDb.cpp.orig	2004-06-10 17:43:42.000000000 +0200
+++ qtstalker-0.26/lib/ChartDb.cpp	2004-06-10 17:46:40.000000000 +0200
@@ -42,9 +42,20 @@
 
 int ChartDb::openChart (QString path)
 {
+#if (DB_VERSION_MAJOR >= 3)
+	  if (db_create(&db, NULL, 0) != 0)
+	    return TRUE;
+#  if (DB_VERSION_MAJOR == 4) && (DB_VERSION_MINOR >= 1)
+	  if (db->open(db, NULL, (char *) path.latin1(), NULL, DB_BTREE, DB_CREATE, 0664) != 0)
+	    return TRUE;
+#  else
+	  if (db->open(db, (char *) path.latin1(), NULL, DB_BTREE, DB_CREATE, 0664) != 0)
+	    return TRUE;
+#  endif
+	#else
   if (db_open((char *) path.latin1(), DB_BTREE, DB_CREATE, 0664, NULL, NULL, &db) != 0)
     return TRUE;
-  else
+#endif
     return FALSE;
 }
 
